### ISSUE TITLE
fails to build with ghc-9.6 stackage nightly

### ISSUE THREAD

**Issue by juhp** [vincenthz/hs-crypto-random#15]
```
       Building library for crypto-random-0.0.9..                                                                                   
       [3 of 8] Compiling Crypto.Random.Entropy.Unix                                                                                
                                                                                                                                    
       /var/stackage/work/unpack-dir/unpacked/crypto-random-0.0.9-136977cf3aa87f76eaf3b98c983894eb41051f1c5363fcdbd2700bc549b0005f/C
rypto/Random/Entropy/Unix.hs:51:33: error: [GHC-83865]                                                                              
           • Couldn't match expected type: OpenFileFlags -> IO H                                                                    
                         with actual type: IO Fd                                                                                    
           • The function ‘openFd’ is applied to four value arguments,                                                              
               but its type ‘FilePath -> OpenMode -> OpenFileFlags -> IO Fd’                                                        
               has only three                                                                                                       
             In the second argument of ‘fmap’, namely                                                                               
               ‘openFd filepath ReadOnly Nothing fileFlags’                                                                         
             In the first argument of ‘catch’, namely                                                                               
               ‘(Just `fmap` openFd filepath ReadOnly Nothing fileFlags)’                                                           
          |                                                                                                                         
       51 | openDev filepath = (Just `fmap` openFd filepath ReadOnly Nothing fileFlags)                                             
          |                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                              
                                                                                                                                    
       /var/stackage/work/unpack-dir/unpacked/crypto-random-0.0.9-136977cf3aa87f76eaf3b98c983894eb41051f1c5363fcdbd2700bc549b0005f/C
rypto/Random/Entropy/Unix.hs:51:58: error: [GHC-83865]                                                                              
           • Couldn't match expected type ‘OpenFileFlags’                                                                           
                         with actual type ‘Maybe a0’                                                                                
           • In the third argument of ‘openFd’, namely ‘Nothing’                                                                    
             In the second argument of ‘fmap’, namely                                                                               
               ‘openFd filepath ReadOnly Nothing fileFlags’                                                                         
             In the first argument of ‘catch’, namely                                                                               
               ‘(Just `fmap` openFd filepath ReadOnly Nothing fileFlags)’                                                           
          |                                                                                                                         
       51 | openDev filepath = (Just `fmap` openFd filepath ReadOnly Nothing fileFlags)                                             
          |                                                          ^^^^^^^
```

**Comment by andreasabel** [comment]
See: 
- #14

**Comment by ysangkok** [comment]
Attention Yesod users @pbrisbin ([commits](https://github.com/commercialhaskell/stackage/commits?author=pbrisbin)), @mjgpy3 ([commits](https://github.com/commercialhaskell/stackage/commits?author=mjgpy3)), @paul-rouse ([commits](https://github.com/commercialhaskell/stackage/commits?author=paul-rouse)), @ncaq ([commits](https://github.com/commercialhaskell/stackage/commits?author=ncaq)), @bobjflong:

Would it be possible for any of you to either migrate the `clientsession` package to `crypton` or alternatively take over the `crypto-random` package and merge vincenthz/hs-crypto-random#14? If this is not done, Yesod won't be available in Stackage for GHC 9.6.

**Comment by pbrisbin** [comment]
I can take a look at migrating `clientsession` to `crypton`; is there a migration guide anywhere?

**EDIT**: I don't know that I'll be able to do this, that package uses a number of different cryptographic libraries and has various CPP checks around them. Is there a reason you tagged who you did, vs opening Issue on that project itself where hopefully someone more familiar can help out?

**Comment by ysangkok** [comment]
I just tagged people that seemed to be interested in having Yesod on Stackage, judging by their commit history.

The easiest fix would be to patch crypto-random because a patch is already available. That's why I posted it here. One could argue that there strictly speaking isn't a problem with clientsession. Given that there was no existing issue for clientsession, I opted for posting on this existing issue.
